corehq/apps/users/user_data.py

```diff
@@ -63,14 +64,19 @@ def save(self):
 
     @property
     def _provided_by_system(self):
-        return {
+        provided_data = {
             **(self.profile.fields if self.profile else {}),
             PROFILE_SLUG: self.profile_id or '',
             COMMCARE_PROJECT: self.domain,
-            COMMCARE_LOCATION_ID: self._couch_user.location_id,
-            COMMCARE_LOCATION_IDS: self._couch_user.location_ids,
-            COMMCARE_PRIMARY_CASE_SHARING_ID: self._couch_user.location_id,
         }
+        if getattr(self._couch_user, 'location_id', None):
```

Comment:
It occurs to me that this is a difference between web and mobile workers - for web, the fields are always present, but for mobile, they're only present if non-empty.  That is a gap - I didn't consider the significance of that distinction when implementing the web users bit, but I think we should bring it in to line.  If that makes sense to do as part of this PR, great, otherwise I can follow up with it afterwards.

Reply:
Hi Ethan, what do you mean by "bring it in to line"? I saw `CouchUser` has `get_location_id` , and both `CommCareUser` and `WebUser` have `get_location_ids`. I will just call this two methods instead of access the fields directly. What else do you want me to include in this PR?

Reply:
That should be correct, thank you.  I mean that right now web users always have `commcare_location_id` in their restores, even when not assigned a location, because of that first block of code I linked above.  This I now think is an inconsistency, and I'd like to give web users the same behavior as mobile workers.  Happy to do that in another PR though.

Reply:
Thank you Ethan!
- Will make `comcare_location_id` only present if non-empty have an unexpected impact for web user?
- "web users don't have a assigned_location_ids attribute" > I see we don't store `location_id` and `assigned_location_ids` for `WebUser`, but we can still access them on `WebUser` because they are attributes for `CouchUser`, it is just empty. Would move them to `CommCareUser` make sense?
- I made changes in 911410a3d3ad80abb721be05449dc52ecd255fc1 and 5193c764639da7e6468172de0c06db4d3278d53d. But haven't fix the inconsistency. I would prefer having that fix in a separate PR and make sure it is safe to make that change.

Reply:
Ethan, my change failed on `test_user_data_ignores_location_fields` (https://github.com/dimagi/commcare-hq/actions/runs/9912924492/job/27388787564?pr=34852). Seems like previous behavior is, WebUser don't have `commcare_location_id` and `commcare_location_ids` in their user data! Now after my change, it will has... I'm concerned of change the existing behavior. Also I don't know if I by accident achieved the "bring it in to line"... I might still not fully understand what you mean and what you're trying to achieve. 😞

Reply:
Ethan, my change failed on `test_user_data_ignores_location_fields` (https://github.com/dimagi/commcare-hq/actions/runs/9912924492/job/27388787564?pr=34852). Seems like previous behavior is, WebUser don't have `commcare_location_id` and `commcare_location_ids` in their user data! Now after my change, it will has... I'm concerned of change the existing behavior. Also I don't know if I by accident achieved the "bring it in to line"... I might still not fully understand what you mean and what you're trying to achieve. 😞

Reply:
Ethan, my change failed on `test_user_data_ignores_location_fields` (https://github.com/dimagi/commcare-hq/actions/runs/9912924492/job/27388787564?pr=34852). Seems like previous behavior is, WebUser don't have `commcare_location_id` and `commcare_location_ids` in their user data! Now after my change, it will has... I'm concerned of change the existing behavior. Also I don't know if I by accident achieved the "bring it in to line"... I might still not fully understand what you mean and what you're trying to achieve. 😞

Reply:
Yeah, for web users, the location fields are only available in the restore and user cases, not in user data directly.  I think for this PR it should be sufficient to add a check to only insert them in user data for mobile workers, but I'd still go through the `get_location_id` type accessors.